spurious failure in test_cache_files

current master branch on travis-ci:
```
_________________ ArchiverCorruptionTestCase.test_cache_files __________________

[gw0] linux -- Python 3.8.6 /home/travis/build/borgbackup/borg/.tox/py38/bin/python

self = <borg.testsuite.archiver.ArchiverCorruptionTestCase testMethod=test_cache_files>
    def test_cache_files(self):
        self.cmd('create', self.repository_location + '::test', 'input')
        self.corrupt(os.path.join(self.cache_path, 'files'))
        out = self.cmd('create', self.repository_location + '::test1', 'input')
        # borg warns about the corrupt files cache, but then continues without files cache.
>       assert 'files cache is corrupted' in out
E       AssertionError: assert 'files cache is corrupted' in ''
.tox/py38/lib/python3.8/site-packages/borg/testsuite/archiver.py:3714: AssertionError
```

Bounty: https://www.bountysource.com/issues/93666717-spurious-failure-in-test_cache_files

--- a/src/borg/testsuite/archiver.py
+++ b/src/borg/testsuite/archiver.py
@@ -3686,10 +3686,12 @@ class ArchiverCorruptionTestCase(ArchiverTestCaseBase):
         self.cmd('init', '--encryption=repokey', self.repository_location)
         self.cache_path = json.loads(self.cmd('info', self.repository_location, '--json'))['cache']['path']
 
-    def corrupt(self, file):
+    def corrupt(self, file, amount=1):
         with open(file, 'r+b') as fd:
-            fd.seek(-1, io.SEEK_END)
-            fd.write(b'1')
+            fd.seek(-amount, io.SEEK_END)
+            corrupted = bytes(255-c for c in fd.read(amount))
+            fd.seek(-amount, io.SEEK_END)
+            fd.write(corrupted)
 
     def test_cache_chunks(self):
         self.corrupt(os.path.join(self.cache_path, 'chunks'))

